proposals/threads/Overview.md

```diff
@@ -1 +1,480 @@
+| `i32.atomic.rmw16_u.cmpxchg` | 2 bytes | `expected` wrapped from i32 to i16, 16-bit compare equal | wrapped from i32 to i16, store 2 bytes | zero-extended from i8 to i32 |
+| `i32.atomic.rmw.cmpxchg` | 4 bytes | 32-bit compare equal | store 4 bytes | as i32 |
+| `i64.atomic.rmw8_s.cmpxchg` | 1 byte | `expected` wrapped from i64 to i8, 8-bit compare equal | wrapped from i64 to i8, store 1 byte | sign-extended from i8 to i64 |
+| `i64.atomic.rmw8_u.cmpxchg` | 1 byte | `expected` wrapped from i64 to i8, 8-bit compare equal | wrapped from i64 to i8, store 1 byte | zero-extended from i8 to i64 |
+| `i64.atomic.rmw16_s.cmpxchg` | 2 bytes | `expected` wrapped from i64 to i16, 16-bit compare equal | wrapped from i64 to i16, store 2 bytes | sign-extended from i16 to i64 |
+| `i64.atomic.rmw16_u.cmpxchg` | 2 bytes | `expected` wrapped from i64 to i16, 16-bit compare equal | wrapped from i64 to i16, store 2 bytes | zero-extended from i16 to i64 |
+| `i64.atomic.rmw32_s.cmpxchg` | 4 bytes | `expected` wrapped from i64 to i32, 32-bit compare equal | wrapped from i64 to i32, store 4 bytes | sign-extended from i32 to i64 |
+| `i64.atomic.rmw32_u.cmpxchg` | 4 bytes | `expected` wrapped from i64 to i32, 32-bit compare equal | wrapped from i64 to i32, store 4 bytes | zero-extended from i32 to i64 |
+| `i64.atomic.rmw.cmpxchg` | 8 bytes | 64-bit compare equal | 8 bytes | as i64 |
+
+### Lock-free Guarantees
+
+Only 32-bit atomic accesses are guaranteed to be lock-free. This matches the
+[ECMAScript specification](https://tc39.github.io/ecma262/#sec-atomics.islockfree).
+
+### Alignment
+
+Unlike normal memory accesses, misaligned atomic accesses trap. For non-atomic
+accesses on shared linear memory, misaligned accesses do not trap.
+
+It is a validation error if the alignment field of the memory access immediate
+has any other value than the natural alignment for that access size.
+
+## Thread operators
+
+One new operator is added: `is_lock_free`. It is an optimization primitive.
+High-performance algorithms will use `is_lock_free` to determine whether to use
+locks or atomic operations in critical sections.
+
+  * `is_lock_free`: given an operand `N` of type `i32`, if the atomic step of an
+    atomic primitive (see [Atomic Memory Accesses](#atomic-memory-accesses)) on
+    a datum of size `N` bytes will be performed without the [agent][] acquiring
+    a lock outside the `N` bytes comprising the datum, then return `1`.
+    Otherwise, return `0`. Once the value of `is_lock_free` for a given value
+    `N` has been observed for any [agent][] in an [agent cluster][], it cannot
+    change.
+
+## Wait and Wake operators
+
+The wake and wait operators are optimizations over busy-waiting for a value to
+change. It is a validation error to use these operators on non-shared linear
+memory. The operators have sequentially consistent ordering.
+
+Both wake and wait operators trap if the effective address of either operator
+is misaligned or out-of-bounds.
```

Comment:
"Misaligned" for `wake` means 4 byte aligned, right? This should be spelled out, or perhaps `wake` shouldn't care about alignment.